src/shorah/shorah_snv.py

```diff
@@ -400,23 +410,82 @@ def main(args):
         write_list.append(s.rstrip().split('\t'))
         x += 1
 
-    # sort p values, correct with Benjamini Hochberg and write to file
+    # sort p values, correct with Benjamini Hochberg and append to output
     p_vals_m.sort()
     q_vals = BH(p_vals_m, len(p_vals_m))
-    csv_file = '.'.join(snpFile.split('.')[:-1]) + '_final.csv'
-    with open(csv_file, 'w') as cf:
-        writer = csv.writer(cf)
+    for q, i3 in q_vals:
+        write_list[i3].append(q)
+
+
+    # Write ShoRAH csv output file
+    if 'csv' in args.format:
+        csv_file = '{}_final.csv'.format(os.path.splitext(snpFile)[0])
         if increment == 1:
             header_row = ['Chromosome', 'Pos', 'Ref', 'Var', 'Freq', 'Post',
                           'Fvar', 'Rvar', 'Ftot', 'Rtot', 'Pval', 'Qval']
         else:
             header_row = ['Chromosome', 'Pos', 'Ref', 'Var', 'Frq1', 'Frq2',
                           'Frq3', 'Pst1', 'Pst2', 'Pst3', 'Fvar', 'Rvar',
                           'Ftot', 'Rtot', 'Pval', 'Qval']
-        writer.writerow(header_row)
-        for q, i3 in q_vals:
-            write_list[i3].append(q)
-        # only print when q >= 5%
-        for wl in write_list:
-            if wl[-1] >= 0.05:
-                writer.writerow(wl)
+        with open(csv_file, 'w') as cf:
+            writer = csv.writer(cf)
+            writer.writerow(header_row)
+            # only print when q >= 5%
+            for wl in write_list:
+                if wl[-1] >= 0.05:
+                    writer.writerow(wl)
+
+    # Write VCF output file
```

Comment:
Using f-strings instead of `str.format` might make this more concise.

Reply:
Adjusted. Python 3.6 is the required one